client/src/components/Navbar.js

```diff
@@ -26,25 +31,73 @@ const useStyles = makeStyles((theme) => ({
   }));
 
 let needsFetchingLists = true;
-let needsCleanList = false;
+let msgHasBeenRead = false;
+let needsSetSocket = true;
 
 const Navbar = ()=>{
     const classes = useStyles();
     const authContext = useContext(AuthContext);
     const shListsContext = useContext(ShListsContext);
+    const [notification, setNotification] = useState({messages:[]});
+    const [newMsg, setNewMsg] = useState(null);
+    const [socket, setSocket] = useState({socket:null})
     
 
     useEffect(() => {
-        if(authContext.isAuthenticated && needsFetchingLists){ //pretects, because private route is uncommented for developing
+        if(authContext.isAuthenticated && needsFetchingLists){ 
             fetchShLists(shListsContext.dispatchShLists,shListsContext.handleShListsFailure);
             needsFetchingLists = false;
-            console.log("Navbar/useEffet is fetching");
         }
-        if(needsCleanList){
-            shListsContext.handleShListsFailure({response:null});
-            needsCleanList = false;
+        if(authContext.isAuthenticated && needsSetSocket){
+            const socket = socketIOClient(ENDPOINT);
+            needsSetSocket = false;
+
+            socket.on('price_notification', data => {
+                setNewMsg(data.message);
+                msgHasBeenRead = false;
+            });
+            socket.emit('join_room', {
+                message: "",
+                userId: authContext._id,
+            });
+            setSocket({socket:socket});
+        }
+        if(newMsg && newMsg !== null){
```

Comment:
is it redundant here to check for both conditions?

Reply:
Oh! I will clean it.